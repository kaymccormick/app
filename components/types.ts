import Entity from '../model/entity/Entity';
import * as Model from '../model';
import {List} from "immutable";
import {DropTargetMonitor} from "react-dnd";
import {ApplicationState, EntityUIState} from "../model/types";
import { SiteInterface } from '../src/types';
import { WebApplication }from '../src/WebApplication'

export interface MainComponentBaseProps {
site?: SiteInterface;
app?: WebApplication;
}

export interface EntityProps {
    entity?: Model.Entity;
    editMode?: boolean;
    // @ts-ignore
    connectDragSource?: any;
    isDragging?: boolean;
    modelKey: string;
    x?: number;
    y?: number;
    width?: number;
    height?: number;
    index: number;
    updateEntity?: any;
    changeSelection?: any;
}

export interface Rect {
    x: number;
    y: number;
    width: number;
    height: number;
}

export interface EntityDragDropItem {
    modelKey: string;
    clientRect?: Rect;
}

export interface EntityViewProps {
    entities: List<Entity>;
    entityUIState: List<EntityUIState>;
    selectedIndex?: number | undefined;
    canDrop?: boolean;
    connectDropTarget?: any;
    handleDrop?: (monitor: DropTargetMonitor) => void;
    moveEntity?: (index: number, x: number, y: number) => ApplicationState;
}
export interface EntityContainerProps {
    index?: number;
    entity?: Model.Entity;
    updateEntity?: any;
    modelKey?: string;
}
export interface EntityViewContainerProps {
}
export interface EntitySectionProps {
    section: string;
    sectionTitle?: string;
    index?: number;
    entity?: Model.Entity;
}
export interface EntityAttributesSectionProps extends EntitySectionProps {
// @ts-ignore
    addAttribute?: any;
}

export interface EntityMethodsSectionProps extends EntitySectionProps {
}
export interface MyEditorState {
}
export const Types = {
    ENTITY: 'entity',
}
export interface EntityAttributeProps {
    entityAttribute: Model.EntityAttribute;
}
