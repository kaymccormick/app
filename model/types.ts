import {ADD_ATTRIBUTE, ADD_ENTITY, MOVE_ENTITY} from './actions';
import {List} from "immutable";
import Entity from "./Entity";

export type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;

export interface Pojo {[propName: string]: any}
export interface Action {
    type: string;
}
export interface MoveEntityAction extends Omit<Action, 'type'> {
    type: "MOVE_ENTITY";
    index: number;
    x: number;
    y: number;
}
export interface AddEntityAction extends Omit<Action, 'type'> {
    type: "ADD_ENTITY";
}
export interface ChangeSelectionAction extends Omit<Action, 'type'> {
type: "CHANGE_SELECTION";
index: number;
}

export interface EntityUIState {
    x?: number;
    y?: number;
}

export interface EntityViewUIState {
    selectedIndex: number | undefined;
    entityUIState: List<EntityUIState>;
}

export interface ModelState {
    entities: List<Entity>;
}

export interface ApplicationState {
    model: ModelState;
    entityViewUI: EntityViewUIState;
}
export interface AddAttributeAction extends Omit<Action, 'type'> {
type:"ADD_ATTRIBUTE";
  index:number;
  }
export interface AddMethodAction extends Omit<Action, 'type'> {
type:"ADD_METHOD";
  index:number;
  }

export interface UpdateEntityAction extends Omit<Action, 'type'> {
type:"UPDATE_ENTITY";
  index:number;
  entityData: Entity;
  }



export interface EntityViewState {
x?: number;
y?: number;
}
