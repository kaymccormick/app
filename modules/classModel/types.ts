import { Map, List } from 'immutable';
import { ApplicationModule } from '../../src/ApplicationModule';
import {Pojo} from 'classModel';

export interface InputObject {
    [propName: string]: any;
}
export const REQUEST_PROJECT_DATA = 'REQUEST_PROJECT_DATA';
export const LOAD_PROJECT_DATA= 'LOAD_PROJECT_DATA';
export const RECEIVE_PROJECT_DATA = 'RECEIVE_PROJECT_DATA';
export const FETCH_PROJECT_DATA = 'FETCH_PROJECT_DATA';
export const REQUEST_INITIAL_DATA = 'REQUEST_INITIAL_DATA';
export const FETCH_INITIAL_DATA = 'FETCH_INITIAL_DATA';
export const LOAD_INITIAL_DATA = 'LOAD_INITIAL_DATA';
export const RECEIVE_INITIAL_DATA = 'RECEIVE_INITIAL_DATA';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface Module extends ApplicationModule<ModuleState> {
}

export interface RequestProjectDataAction {
    type: typeof REQUEST_PROJECT_DATA;
}
export interface RequestInitialDataAction {
    type: typeof REQUEST_INITIAL_DATA;
}
export interface ReceiveInitialDataAction {
    type: typeof RECEIVE_INITIAL_DATA;
    result: InputObject;
}
export interface LoadInitialDataAction {
    type: typeof LOAD_INITIAL_DATA;
}
export interface FetchInitialDataAction {
    type: typeof FETCH_INITIAL_DATA;
}
export interface LoadProjectDataAction {
    type: typeof LOAD_PROJECT_DATA;

}
export interface ReceiveProjectDataAction {
    type: typeof RECEIVE_PROJECT_DATA;
    projects: List<Pojo.ProjectPojo>;
}

export type ActionTypes = ReceiveProjectDataAction | LoadProjectDataAction | RequestProjectDataAction | RequestInitialDataAction |
ReceiveInitialDataAction | LoadInitialDataAction | FetchInitialDataAction;

export type EntitiesType = Map<string, Map<number, Pojo.BasePojo>> ;
export interface ModuleState {
/* Module state fields */
    entities: EntitiesType;
    projects: List<Pojo.ProjectPojo>;
}
