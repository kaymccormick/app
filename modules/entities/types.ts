export interface ReceiveEntitiesAction {
    type: string;
    entities: EntityPojo[];
}
export interface EntitiesState {
    entities?: EntityPojo[];
}

export interface EntityPojo {
    name?: string;
    tableName?: string;
    columns?: EntityColumnPojo[];
}
export interface EntityColumnPojo {
    propertyName?: string;
    type?: string;
    isPrimary?: boolean;
    isNullable?: boolean;
    isGenerated?: boolean;
    comment?: string;
    isArray?: boolean ;
    propertyPath?: string;
    isObjectId?: boolean;
}
