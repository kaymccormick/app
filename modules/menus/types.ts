import { Map, List } from 'immutable';
import { ApplicationModule } from '../../src/ApplicationModule';

export const POPULATE_MENUS = 'POPULATE_MENUS';
export const ADD_MENU_ITEM = 'ADD_MENU_ITEM';
export const SELECT_ITEM = 'SELECT_ITEM';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface Module extends ApplicationModule<ModuleState> {
}

export interface PopulateMenusAction {
  type: typeof POPULATE_MENUS;
}

export interface AddMenuItemAction {
  type: typeof ADD_MENU_ITEM;
  item: MenuItemPojo;
}
export interface SelectItemAction {
  type: typeof SELECT_ITEM;
  item: MenuItemPojo;
}

export type ActionTypes = PopulateMenusAction  | AddMenuItemAction | SelectItemAction /* | other types */

export interface MenuItemPojo {
    parentKey?: string;
    key: string;
    title: string;
    subItems?: Set<string>;
}

export interface ModuleState {
/* Module state fields */
/* After adding a field here, it needs to be returned from
   Module.getInitialState */
    menuItems: Map<string, MenuItemPojo>;
}
