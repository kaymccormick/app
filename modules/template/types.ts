import { Map, List } from 'immutable';
import { ApplicationModule } from '../../src/ApplicationModule';

export const ACTION_TYPE = 'ACTION_TYPE';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface Module extends ApplicationModule<ModuleStaate> {
}

export interface ActionTypeAction {
    type: typeof ACTION_TYPE;
    /* other fields */
}
/* other action typs */

export type ActionTypes = ActionTypeAction /* | other types */

export interface ModuleState {
/* Module state fields */
}
