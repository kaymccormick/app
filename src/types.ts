import { ApplicationModule } from './ApplicationModule';
import Tree from '../model/tree/Tree';
import { AppLogger } from './AppLogger';
import { Configuration } from './Configuration';
import { WebApplication } from './WebApplication';

export type ModuleType = ApplicationModule<any>;

export interface BasicLogger {
    debug: LeveledLogMethod;
    error: LeveledLogMethod;
}

type LogCallback = (error?: any, level?: string, message?: string, meta?: any) => void;
export interface LeveledLogMethod {
    (message: string, callback: LogCallback): any;
    (message: string, meta: any, callback: LogCallback): any;
    (message: string, ...meta: any[]): any;
//    (infoObject: object): any;
}

export interface JsConfig {
    [propName: string]: any;
}


// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface Resources { [resourceName: string]: any }

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface Reducers { [reducerName: string]: any }

export type CoreModuleKey = "menus"
| "entities"
| "classModel"
| "jsontree"
| string
;

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type CoreModule = ApplicationModuleInterface<any>;

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type ApplicationModuleType = ApplicationModule<any>;

export interface Action {
    type: string;
}

export interface SiteInterface {
    mainMenuTree: Tree;
}

export interface Actions {
// eslint-disable-next-line @typescript-eslint/no-explicit-any
    [actionName: string]: any;
}

export interface ApplicationModuleInterface<S> {
    logger: AppLogger;
    actions: Actions;
    name: string;
    key: string;
    setup(app: WebApplication, config: Configuration): void;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    getReducers(): any;
    getInitialState(): S;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    getMainComponent(): Promise<any>;
}
