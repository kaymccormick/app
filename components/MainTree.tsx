import React from 'react';
import Tree, { TreeNode } from 'rc-tree';
import {SiteInterface} from '../src/types';
import RcTreeAdapter from '../model/tree/RcTreeAdapter';
import {EntitiesState} from '../modules/entities/types';

interface MainTreeState {
    autoExpandParent?: any;
    expandedKeys?: any;
    treeData?: any;
}

export interface MainTreeProps {
    site?: SiteInterface;
    entities?: EntitiesState,
}

export default class MainTree extends React.Component<MainTreeProps> {
    state: MainTreeState = {};
    onExpand: (expandedKeys: {}) => void;
  
    constructor(props: MainTreeProps) {
        super(props);
        if(props.site) {
        const adapter = new RcTreeAdapter(props.site.mainMenuTree);
        this.state = { treeData: adapter.toData() };
  }
        this.onExpand = this._onExpand.bind(this);
    }

    _onExpand(expandedKeys: {}) {
        this.setState({expandedKeys, autoExpandParent: false });
    }

    render() {
        // @ts-ignore
        const loop = data => {
            // @ts-ignore
            return data.map((item) => {
                if (item.children) {
                    return (
                        <TreeNode
                            key={item.key} title={item.title}
                            disableCheckbox={item.key === '0-0-0-key'}
                        >
                            {loop(item.children)}
                        </TreeNode>
                    );
                }
                return <TreeNode key={item.key} title={item.title} />;
            });
        };
        return <Tree
            onExpand={this.onExpand}
            expandedKeys={this.state.expandedKeys}
            autoExpandParent={this.state.autoExpandParent}>
            {loop(this.state.treeData)}
        </Tree>;
    }
}
