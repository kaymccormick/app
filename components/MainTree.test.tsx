test.todo('');
/*import React from 'react';
import { shallow,mount } from 'enzyme';
import MainTree from './MainTree';
import { EntitiesState,EntityPojo,EntityColumnPojo } from '../modules/entities/types';

// @ts-ignore
const entitiesState:  EntitiesState = { entities: [{name:' test'}] };

test.skip('1', () => {
const wrapper = mount(<MainTree entities={entitiesState}/>);
const r = wrapper.find('Tree');
expect(r.html()).toMatchSnapshot();
});


*/