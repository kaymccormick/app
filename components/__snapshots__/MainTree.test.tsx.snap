// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 1`] = `"<ul class=\\"rc-tree\\" role=\\"tree\\" unselectable=\\"on\\"><li class=\\"rc-tree-treenode-switcher-close\\" role=\\"treeitem\\"><span class=\\"rc-tree-switcher rc-tree-switcher_close\\"></span><span title=\\"Entities\\" class=\\"rc-tree-node-content-wrapper rc-tree-node-content-wrapper-close\\"><span class=\\"rc-tree-iconEle rc-tree-icon__close\\"></span><span class=\\"rc-tree-title\\">Entities</span></span></li></ul>"`;
